meta-facebook/yv3-vf/src/platform/plat_power_seq.c

```diff
@@ -0,0 +1,305 @@
+{
+	uint8_t i;
+
+	for (i = 0; i < M2_IDX_E_MAX; i++) {
+		if (get_fm_p12v_sw_en(i))
+			break;
+	}
+
+	if (i == M2_IDX_E_MAX) {
+		dev_12v_fault_hander(); // control PWRGD_EXP_PWROK & LED_PWRGD_P12V_E1S_ALL
+		gpio_set(FM_P12V_EDGE_EN, 0);
+		gpio_set(FM_CLKBUF_EN, 0);
+	}
+}
+
+/*
+ * +--------+---------------+------------------------------------------+
+ * |        | AC            | ON                                       |
+ * |        +---------------+----------------+-------------------------+
+ * |        | DC            | OFF            | ON                      |
+ * |        +---------------+-------+--------+-------+--------+--------+
+ * |        | PRSNT         | ON    | OFF    | ON    | ON     | OFF    |
+ * |        +---------------+-------+--------+-------+--------+--------+
+ * |        | DRIVE PWR     | OFF            | ON    | OFF    | OFF    |
+ * +--------+---------------+-------+--------+-------+--------+--------+
+ * | FM_P12V_E1S_X_EN       | L     | L      | H     | L      | L      |
+ * +------------------------+-------+--------+-------+--------+--------+
+ * | FM_P3V3_E1S_X_SW_EN    | H     | L      | H     | H      | L      |
+ * +------------------------+-------+--------+-------+--------+--------+
+ * | CLKBUF_E1S_X_OE_N      | H     | H      | L     | L      | H      |
+ * +------------------------+-------+--------+-------+--------+--------+
+ */
+
+uint8_t m2_dev_power_switch_with_pwrdis_chk(uint8_t idx, uint8_t enable, uint8_t chk_pwrdis,
+					    uint8_t force_ctl_3v3)
+{
+	static uint8_t is_force_disable_3v3[M2_IDX_E_MAX];
+
+	const uint8_t dc_en = gpio_get(FM_POWER_EN);
+	const uint8_t hsc_pwrgd = get_e1s_pwrgd();
+	const uint8_t prsnt = m2_prsnt(idx);
+
+	const uint8_t en_12v = dc_en && hsc_pwrgd && prsnt && enable;
+	const uint8_t en_3v3 = force_ctl_3v3 ? enable : prsnt;
+	const uint8_t en_clk = enable && prsnt;
+
+	/* if the PWRDIS is enable by user, don't control the drive power by 12V/3V3 SW. */
+	if (chk_pwrdis) {
+		if (get_fm_pwrdis_status(idx))
+			return 1;
+	}
+
+	/* for the drive power off, the 12v should be disable before 3v3 */
+	if (!enable) {
+		clkbuf_oe_en(idx, en_clk);
+		delay_function(3, fm_p12v_sw_en, idx, en_12v);
+	}
+
+	if (force_ctl_3v3 && (enable == 0)) {
+		is_force_disable_3v3[idx] = 1;
+		fm_p3v3_sw_en(idx, en_3v3);
+	} else if (force_ctl_3v3 && (enable == 1)) {
+		is_force_disable_3v3[idx] = 0;
+	}
+
+	// check whether force disable
+	if (!is_force_disable_3v3[idx])
+		fm_p3v3_sw_en(idx, en_3v3);
+
+	if (enable) {
+		fm_p12v_sw_en(idx, en_12v);
+
+		//delay 40 ms to enable clkbuf after enable 12V
+		delay_function(50, clkbuf_oe_en, idx, en_clk);
+	}
+
+	if (!dc_en) {
+		delay_function(20, check_dc_off_process, 0, 0);
+	}
+
+	return 0;
+}
+
+uint8_t m2_dev_power_switch(uint8_t idx, uint8_t enable)
+{
+	return m2_dev_power_switch_with_pwrdis_chk(idx, enable, 1, 0);
+}
+
+uint8_t device_all_power_set(uint8_t idx, uint8_t set_val)
+{
+	uint8_t is_on = (set_val & DEV_PWR_ON ? 1 : 0);
```

Comment:
Please fix the linter issues in lines 227, 228 and 229.

Reply:
fix it on #469

Reply:
change to 96bcc87237f15997b4a166554376232bd49cd73e